File: 프로그래머스/2/140107. 점 찍기/점 찍기.py
import math

def solution(k, d):
    answer = 0
    
    for x in range(0, d+1, k): # d 거리 내에서 가능한 x 좌표
        y = math.isqrt(d**2 - x**2)
        answer += y // k + 1
        
    return answer

File: 프로그래머스/2/140107. 점 찍기/점 찍기.java
import java.util.*;
import java.lang.Math;

class Solution {
    public long solution(int k, int d) {
        long answer = 0;
        
        for (int x = 0; x < d+1; x += k) {
            int y = (int) Math.sqrt(Math.pow(d, 2) - Math.pow(x, 2));
            answer += y / k + 1;
        }
        return answer;
    }
}